src/generate/renders/tests/CliTestPrepare.ts

```diff
@@ -1,12 +1,32 @@
 import * as path from 'path';
 import { CodeModelAz } from '../../CodeModelAz';
-import { HeaderGenerator } from '../Header';
 import { TemplateBase } from '../TemplateBase';
 import { PathConstants } from '../../../utils/models';
+import { GenPreparerName, preparerInfos } from './ScenarioTool';
+import { ToJsonString } from '../../../utils/helper';
+
+class FuncData {
+    constructor(public cmdTemplate: string, public cmdFormats: string[]) {}
+}
+export class PreparerData {
+    public initParamPairs: [string, string][];
+    public attributes: string[];
+    public createFunc: FuncData;
+    public removeFunc: FuncData;
+
+    constructor(public className: string) {
+        this.initParamPairs = [];
+        this.attributes = [];
+        this.createFunc = new FuncData(undefined, []);
+        this.removeFunc = new FuncData(undefined, []);
+    }
+}
 
 export class CliTestPrepare extends TemplateBase {
-    constructor(model: CodeModelAz) {
+    resourceNames: string[];
+    constructor(model: CodeModelAz, resourceNames: string[]) {
         super(model);
+        this.resourceNames = resourceNames;
         if (this.model.IsCliCore) {
```

Comment:
```
            this.relativePath = path.join(
                model.AzextFolder,
                PathConstants.testFolder,
                PathConstants.latestFolder,
                PathConstants.preparersFile,
            );
```
   I think you don't need the if (model.IsCliCore) condition any more. this expression will return the right path in both extension mode and core mode